aeron-cluster/src/main/java/io/aeron/cluster/ConsensusControlState.java

```diff
@@ -16,81 +16,24 @@
 package io.aeron.cluster;
 
 import io.aeron.ExclusivePublication;
-import io.aeron.Subscription;
 
 /**
  * The state needed to allow control of the consensus module.
  * <p>
  * This is a record object being passed to external entities.
  *
+ * @param logPublication        publication or null.
+ * @param logRecordingId        log recording id.
+ * @param leadershipTermId      leadership term id.
  * @see ConsensusModuleExtension
  */
-public final class ConsensusControlState
+public record ConsensusControlState(ExclusivePublication logPublication, long logRecordingId, long leadershipTermId)
 {
```

Comment:
One thing that definitely is causing binary incompatibility is dropping the leaderLogSubscription. The getter exists in the old code, but doesn't exist in the new code.